_posts/2023-05-25-Announcing-Data-Prepper-2.3.0.md

```diff
@@ -0,0 +1,147 @@
+
+* String expressions
+    * String expressions evaluate to a result of string type. The string concatenation operator is supported in addition to using functions or JSON pointers or literals as operands. For example, the following expression adds the message1 field of type string in the event with message2 field of type string in the event and appends "suffix" to it.
+
+```/message1 + /message2 + "suffix"```
+
+## Event tagging
+
+Data Prepper 2.3 supports tagging events while using the `grok` processor. Events can be tagged optionally using the following configuration
+
+```
+processor:
+   - grok: 
+        match:
+          message: <pattern>
+        tags_on_match_failure: ["grok_match_fail", "another_tag"]
+```
+
+The presence of tags can be checked in conditional expressions in different processors or routing using `hasTags()` function to do conditional ingestion. For example, a conditional expression checking for the presence of tag `grok_match_fail` would be `hasTags("grok_match_fail")`
+
+## Enhancements to `add_entries` processor
+
+### Expression-based value
+
+The `add_entries` processor is enhanced to support adding values based on an expression where the return type can be boolean, integer or float, or string. The following example shows an `add_entries` processor configuration with a `value_expression` option:
+
+```
+processor:
+   - add_entries:
+       entries:
+         - key: "request_len"
+           value_expression: "length(/request)"
+```
+
+This configuration adds an entry with the key `request_len` with value equal to the length of the request field in the key. The value expression can be any of the expressions supported. See [Supported operators](https://github.com/opensearch-project/data-prepper/blob/main/docs/expression_syntax.md) for more details.
+
+### Setting event metadata keys
+
+The `add_entries` processor can add entries in the event’s metadata instead of the event itself. The following example shows an `add_entries` processor configuration for adding an entry to metadata:
+
+```
+processor:
+   - add_entries:
+       entries:
+         - metadata_key: "request_len"
+           value_expression: "length(/request)"
+```
+This configuration adds an attribute to event metadata with the attribute key `request_len` with value equal to the length of the request field in the key. The value can be set using the value, format, or `value_expression` option of entries field.
+
+
+## Amazon S3 sink
+
+Data Prepper now supports saving data to Amazon S3 sinks as Newline delimited JSON (ndjson). Amazon S3 is a popular choice for storing large volumes of data reliably and in a cost-effective way.
+
+Ingesting data into Amazon S3 offers many possibilities for your data pipelines, including some of the following:
+
+* Noisy or uninteresting data can be routed to Amazon S3 and not to OpenSearch to reduce load on your OpenSearch cluster. This can help you save on compute and storage costs.
```

Comment:
```suggestion
* Noisy or uninteresting data can be routed to Amazon S3 and not to OpenSearch in order to reduce load on your OpenSearch cluster. This can help you save on compute and storage costs.
```